src/main/kotlin/org/kamiblue/client/gui/rgui/component/StringButton.kt

```diff
@@ -18,27 +18,27 @@ class StringButton(val setting: StringSetting) : BooleanSlider(setting.name, 1.0
 
     override fun onStopListening(success: Boolean) {
         if (success) {
-            setting.setValue(name)
+            setting.setValue(componentName)
```

Comment:
Why are you using a getter here?

Reply:
1) it's a setter
2) psure that function; it's just named that way and it does other stuff as well

Reply:
no, for string settings it just 
```
override fun setValue(valueIn: String) {
        value = valueIn
    }
```

Sorry for the mistake in my terminology, I just wrote the wrong thing down.

Reply:
It is a function defined in AbstractSetting so any type of setting can be changed with string. Like set command or parsing from json string